src/webgpu/shader/validation/expression/call/builtin/normalize.spec.ts

```diff
@@ -0,0 +1,168 @@
+const builtin = 'normalize';
+export const description = `
+Validation tests for the ${builtin}() builtin.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf, objectsToRecord } from '../../../../../../common/util/data_tables.js';
+import {
+  Type,
+  kConcreteIntegerScalarsAndVectors,
+  kConvertableToFloatVectors,
+  scalarTypeOf,
+} from '../../../../../util/conversion.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+import {
+  fullRangeForType,
+  kConstantAndOverrideStages,
+  stageSupportsType,
+  validateConstOrOverrideBuiltinEval,
+} from './const_override_validation.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kValidArgumentTypes = objectsToRecord(kConvertableToFloatVectors);
+
+// These should probably go in a utilities file somewhere?
```

Comment:
Looks like you are just doing fp quanitzation, so I think you should be able to use quantizeToF32 and quantizeToF16 from src/webgpu/util/math.ts

Reply:
You're right! Thanks for pointing out those functions!